api/tests/core/offerers/test_api.py

```diff
@@ -1439,3 +1439,35 @@ def test_get_offerer_stats_dashboard_url():
         "params": {"siren": [offerer.siren], "venueid": [str(venue.id)]},
         "exp": round(time.time()) + 600,
     }
+
+
+class CountOfferersByValidationStatusTest:
+    @override_features(ENABLE_BACKOFFICE_API=True)
+    def test_get_offerer_stats(self, client):
+        # given
+        offerers_factories.UserOffererFactory(offerer__validationStatus=offerers_models.ValidationStatus.NEW)
+        offerers_factories.UserOffererFactory.create_batch(
+            2, offerer__validationStatus=offerers_models.ValidationStatus.PENDING
+        )
+        offerers_factories.UserOffererFactory.create_batch(
+            3, offerer__validationStatus=offerers_models.ValidationStatus.VALIDATED
+        )
+        offerers_factories.UserOffererFactory.create_batch(
+            4, offerer__validationStatus=offerers_models.ValidationStatus.REJECTED
+        )
```

Comment:
(suggestion) Est-ce que ça ne vaut pas le coup d'avoir des _factory_ pour ces différents statuts de validation ? J'ai l'impression qu'on joue souvent là-dessus. Après c'est vraiment du détail et un choix très discutable.

Reply:
En fait on a déjà `UserOfferFactory` qui crée par défaut avec la structure en `VALIDATED` ; le repréciser ici est plutôt pour l'homogénéité en comprenant au premier coup d'œil.
Et `UserNotValidatedOffererFactory` qui crée en `NEW`.
L'idée de créer cette dernière pour la distinguer de la première était aussi de gérer en même temps le `validationToken` (voué à disparaître) et le nouveau `validationStatus`.
J'aurais plutôt tendance à dire qu'on pourrait plus tard faire un passage sur les tests pour voir ce qui peut être factorisé, mais plutôt au même moment que la suppression du token (pour Offerer et UserOfferer), histoire de tout remettre au propre en même temps.

Reply:
En plus ce n'est pas si évident de décider le plus clair à lire/comprendre : 
`UserOffererFactory(offerer__validationStatus=PENDING)` => on voit bien qui est en attente
`UserPendingOffererFactory()` => il peut y avoir un doute sur qui est en attente (structure ou rattachement)